packages/ui-kit/src/components/markdown.tsx

```diff
@@ -298,36 +300,64 @@ const TypographyContainer = styled.div`
   ${containerStyles};
 `;
 
+const CopyArea = styled.div`
+  cursor: pointer;
+`;
+
+const TooltipBodyComponent = styled.div`
+  background-color: ${colors.light.surfaceCodeCopy};
+  color: ${colors.light.textInverted};
+  font-weight: ${typography.fontWeights.regular};
+  border-radius: ${tokens.borderRadiusForTooltip};
+  font-size: ${typography.fontSizes.extraSmall};
+  padding: ${dimensions.spacings.xs} ${dimensions.spacings.s};
+`;
+
 /* eslint-disable react/display-name */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const withAnchorLink = (Component: React.ComponentType) => (props: any) => {
-  return (
-    <Component
-      {...props}
-      css={css`
-        display: flex;
-        align-items: baseline;
-        :hover {
-          [data-link-type='anchor-link'] {
-            svg {
-              * {
-                fill: ${colors.light.linkNavigation};
+const withCopyToClipboard =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (Component: React.ComponentType) => (props: any) => {
+    const [isCopiedToClipboard, setIsCopiedToClipboard] = React.useState(false);
+    const handleCopyToClipboardClick = (event: React.SyntheticEvent) => {
+      event.preventDefault();
+      const sectionUrl = `${window.location.href.split('#')[0]}#${props.id}`;
+      copyToClipboard(sectionUrl);
+      setIsCopiedToClipboard(true);
+      setTimeout(() => setIsCopiedToClipboard(false), 1500);
+    };
+    return (
+      <Component
+        {...props}
+        css={css`
+          display: flex;
+          align-items: baseline;
+          :hover {
+            div {
+              svg {
+                * {
+                  fill: ${colors.light.linkNavigation};
+                }
               }
             }
           }
-        }
-        > * + * {
-          margin: 0 0 0 ${dimensions.spacings.s};
-        }
-      `}
-    >
-      <span>{props.children}</span>
-      <a href={`#${props.id}`} data-link-type="anchor-link">
-        <RibbonSvgIcon />
-      </a>
-    </Component>
-  );
-};
+          > * + * {
+            margin: 0 0 0 ${dimensions.spacings.s};
+          }
+        `}
+      >
+        <span>{props.children}</span>
+        <Tooltip
+          title={isCopiedToClipboard ? 'Copied' : 'Copy to clipboard'}
```

Comment:
```suggestion
          title={isCopiedToClipboard ? 'Copied' : 'Copy link to clipboard'}
```
just tuning the language